src/value.jl

```diff
@@ -179,3 +179,11 @@ DataAPI.defaultarray(::Type{CategoricalValue{T, R}}, N) where {T, R} =
   CategoricalArray{T, N, R}
 DataAPI.defaultarray(::Type{Union{CategoricalValue{T, R}, Missing}}, N) where {T, R} =
   CategoricalArray{Union{T, Missing}, N, R}
+
+# define appropriate handlers for JSON3 interface
+StructTypes.StructType(::Type{<:CategoricalValue{<:String}}) = StructTypes.StringType()
```

Comment:
Maybe a single recursive definition like this would be enough? That would cover `AbstractChar`, `Date`, `Bool`, etc.
```julia
 StructTypes.StructType(::Type{<:CategoricalValue{T}}) where {T} =  StructTypes.StructType(T)
```

BTW, @quinnj, we don't support `CategoricalValue(::String)` (nor for any other type) like the StructTypes docs require, so deserialization will probably not work. Is that a problem?

Reply:
When reading in a JSON source it's easy enough to call e.g. `categorical!(df)` on a constructed data frame (e.g. via JSONTables.jl), to get back the categorical representation; so for the simplest approach to the most obvious use case (constructing a dataframe), lacking the read capability isn't a huge problem (and I think making it automatically read categorical values would require some additional tooling in any case).

Reply:
Yeah, we'd only be worried about the read case if someone had something like:
```julia
struct T
    x::CategoricalValue{Float64}
end
```
which I think is pretty rare.

Reply:
It would be much more likely the someone would want to do `JSON3.read(json, CategoricalVector{String})`, so it'd be more worthwhile to figure out if we could support that case, but as @haberdashPI mentioned, it's easy enough to just read stuff in and call `categorical!` afterwards. I mean, the one definition we might consider adding would be:
```julia
StructTypes.construct(::Type{T}, arr) where {T <: CategoricalArray} = categorical!(arr)
```
Which, now that I think about it, might be all that's needed to support reading CategoricalArrays natively. i.e. like if I had a struct like:
```julia
struct T
    id::Vector{Int}
    var::CategoricalVector{String}
end
StructTypes.StructType(::Type{T}) = StructTypes.Struct()
```
@haberdashPI, would you be willing to look into that case?

Reply:
Sure, happy to take a look. I won't get to it until tomorrow...

Reply:
Note that `categorical!` doesn't (and cannot) exist though. Is there a way to allocate directly an array of the right type? Probably not a big deal though, at least for a first implementation.